compiler/syntax/src/res_core.ml

```diff
@@ -2694,24 +2672,24 @@ and parse_jsx p =
   Parser.leave_breadcrumb p Grammar.Jsx;
   let start_pos = p.Parser.start_pos in
   Parser.expect LessThan p;
-  let jsx_expr =
+  let jsx_expr, jsx_attrs =
     match p.Parser.token with
     | Lident _ | Uident _ ->
-      parse_jsx_opening_or_self_closing_element ~start_pos p
+      (parse_jsx_opening_or_self_closing_element ~start_pos p, [jsx_attr])
     | GreaterThan ->
       (* fragment: <> foo </> *)
-      parse_jsx_fragment p
-    | _ -> parse_jsx_name p
+      (parse_jsx_fragment start_pos p, [])
+    | _ -> (parse_jsx_name p, [])
   in
   Parser.eat_breadcrumb p;
-  {jsx_expr with pexp_attributes = [jsx_attr]}
+  {jsx_expr with pexp_attributes = jsx_attrs}
 
 (*
  * jsx-fragment ::=
  *  | <> </>
  *  | <> jsx-children </>
  *)
-and parse_jsx_fragment p =
+and parse_jsx_fragment start_pos p =
```

Comment:
I prefer ranges to be accurate. The location should start at the opening `<` token.